### PR TITLE
test(dm): fix datarace for testSyncerSuite.TestRun

### LINKED ISSUE #4280: DM datarace in testSyncerSuite
### Which jobs are flaking?

https://ci.pingcap.net/blue/organizations/jenkins/atom-ut/detail/atom-ut/8984/tests/

### Which test(s) are flaking?

testSyncerSuite

### Jenkins logs or GitHub Actions link

```log
==================
WARNING: DATA RACE
Read at 0x00c02b781128 by goroutine 453:
  internal/race.Read()
      /usr/local/go/src/internal/race/race.go:37 +0x206
  sync.(*WaitGroup).Add()
      /usr/local/go/src/sync/waitgroup.go:71 +0x219
  github.com/pingcap/tiflow/dm/syncer.(*Syncer).Run()
      /home/jenkins/agent/workspace/atom-ut/tiflow/dm/syncer/syncer.go:1444 +0x395
  github.com/pingcap/tiflow/dm/syncer.(*Syncer).Process()
      /home/jenkins/agent/workspace/atom-ut/tiflow/dm/syncer/syncer.go:670 +0x647
Previous write at 0x00c02b781128 by goroutine 441:
  internal/race.Write()
      /usr/local/go/src/internal/race/race.go:41 +0x125
  sync.(*WaitGroup).Wait()
      /usr/local/go/src/sync/waitgroup.go:128 +0x126
  github.com/pingcap/tiflow/dm/syncer.(*Syncer).Process()
      /home/jenkins/agent/workspace/atom-ut/tiflow/dm/syncer/syncer.go:677 +0x6bc
Goroutine 453 (running) created at:
  github.com/pingcap/tiflow/dm/syncer.(*testSyncerSuite).TestRun()
      /home/jenkins/agent/workspace/atom-ut/tiflow/dm/syncer/syncer_test.go:991 +0x639a
  runtime.call16()
      /usr/local/go/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /usr/local/go/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d
Goroutine 441 (running) created at:
  github.com/pingcap/tiflow/dm/syncer.(*testSyncerSuite).TestRun()
      /home/jenkins/agent/workspace/atom-ut/tiflow/dm/syncer/syncer_test.go:850 +0x3b4d
  runtime.call16()
      /usr/local/go/src/runtime/asm_amd64.s:550 +0x3d
  reflect.Value.Call()
      /usr/local/go/src/reflect/value.go:337 +0xd8
  github.com/pingcap/check.(*suiteRunner).forkTest.func1()
      /go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:850 +0xb3b
  github.com/pingcap/check.(*suiteRunner).forkCall.func1()
      /go/pkg/mod/github.com/pingcap/check@v0.0.0-20200212061837-5e12011dc712/check.go:739 +0x11d
==================
```

### Anything else we need to know

- Does this test exist for other branches as well?

- Has there been a high frequency of failure lately?

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve?

Issue Number: close #4280

### What is changed and how it works?

wait last goroutine `go syncer.Process(ctx, resultCh)` exit befroe start a new one

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

 - Need to update the documentation

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by lance6716 — APPROVED** [review]

File: dm/syncer/syncer_test.go [hunk 1]
```diff
@@ -956,6 +956,7 @@ func (s *testSyncerSuite) TestRun(c *C) {
 	}
 
 	cancel()
+	<-resultCh // wait for the process to finish
```
Comment (WizardXiao) [inline dm/syncer/syncer_test.go:959]: Just a question, can you help to explain some reason about why syncer does not finish when we called `cancel()`.

Comment (Ehco1996) [reply dm/syncer/syncer_test.go:959]: there are  lots of backgraound [jobs ](https://github.com/pingcap/tiflow/blob/4bfa080ecce6c57a670af290a764e2654e9361bb/dm/syncer/syncer.go#L1541)in syncer.Run and they may block main thread exit for little time

**Review by WizardXiao — APPROVED** [review]
LGTM

**Pull request merged**